lib/sth_database.js

```diff
@@ -1841,6 +1841,490 @@ function getNotificationInfo(data, callback) {
+ * @param callback The callback to call with error or the result of the operation
+ */
+function removeDataIfCollection4Attribute(data, options, callback) {
+  var service = data.service,
+      servicePath = data.servicePath,
+      entityId = data.entityId,
+      entityType = data.entityType,
+      attrName = data.attrName;
+  var isAggregated = options.isAggregated;
+
+  var databaseName = getDatabase(service);
+  var database = db.db(databaseName);
+  var collectionName = isAggregated ?
+    getCollectionName4Aggregated(
+      {
+        databaseName: databaseName,
+        servicePath: servicePath,
+        entityId: entityId,
+        entityType: entityType,
+        attrName: attrName
+      }
+    ) :
+    getCollectionName4Events(
+      {
+        databaseName: databaseName,
+        servicePath: servicePath,
+        entityId: entityId,
+        entityType: entityType,
+        attrName: attrName
+      }
+    );
+
+  database.collections(function (err, collections) {
+    if (err) {
+      if (callback) {
+        return process.nextTick(callback.bind(null, err));
+      }
+    }
+
+    var dropCollectionFunctions = [];
+    collections.forEach(function (collection) {
+      if (sthConfig.SHOULD_HASH) {
+        dropCollectionFunctions.push(async.apply(dropHashedCollection, collection.collectionName, data, options));
+      } else if (attrName) {
+        if (collection.collectionName === collectionName) {
+          dropCollectionFunctions.push(async.apply(dropCollection, collection.collectionName, service));
+        }
+      } else if (isAggregated) {
+        if (collection.collectionName.indexOf('.aggr') >= 0 &&
+          collection.collectionName.indexOf(collectionName.slice(0, collectionName.indexOf('undefined'))) >= 0) {
+          dropCollectionFunctions.push(async.apply(dropCollection, collection.collectionName, service));
+        }
+      } else {
+        if (collection.collectionName.indexOf('.aggr') < 0 &&
+          collection.collectionName.indexOf(collectionName.slice(0, collectionName.indexOf('undefined'))) >= 0) {
+          dropCollectionFunctions.push(async.apply(dropCollection, collection.collectionName, service));
+        }
+      }
+    });
+
+    async.parallel(
+      dropCollectionFunctions,
+      callback
+    );
+  });
+}
+
+/**
+ * Removes the raw data associated to the attributes specified in the provided data
+ * @param {object} data It is an object including the following properties:
+ *  - {string} service The service
+ *  - {string} servicePath The service path
+ *  - {string} entityId The entity id
+ *  - {string} entityType The entity type
+ *  - {string} attrName The attribute name. In case no attribute name is provided, all the attributes of the
+ *    provided entity are removed
+ * @param callback The callback to call with error or the result of the operation
+ */
+function removeRawData(data, callback) {
+  switch (DATA_MODEL) {
```

Comment:
DATA_MODEL is used clearly as a configuration parameter, and belongs to the configuration (where it once was). This change should be reverted.